### PR TITLE
Pass an unsigned int by value, rather than reference.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/numerics/kdtree.h [hunk 1]
```diff
@@ -89,7 +89,7 @@ class KDTree
    * your points, and do not call set_points() again, then all following results
    * will likely be wrong.
    */
-  KDTree(const unsigned int &           max_leaf_size = 10,
+  KDTree(const unsigned int             max_leaf_size = 10,
          const std::vector<Point<dim>> &pts = std::vector<Point<dim>>());
```
Comment (masterleinad) [inline include/deal.II/numerics/kdtree.h]: What about `const std::vector<Point<dim>> &pts = {}` while you are here? I am fine with either but we could avoid repeating the longish type and gain a nicer formatting.

Comment (bangerth) [reply include/deal.II/numerics/kdtree.h]: Can I leave this to someone else and another time? I don't have kdtree installed on the laptop I use here at the hackathon.

Comment (masterleinad) [reply include/deal.II/numerics/kdtree.h]: Sure.

**Pull request merged**